### PR TITLE
[#167] add contribution guidelines to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by 2ynn**
@tristanlatr @mina-andajos feel free to edit CONTRIBUTING.md

### CODE REVIEW

**Comment by 2ynn** [comment]
I think a naming convention for Pull Request titles could be good to add as well

File: CONTRIBUTING.md
Comment (mina-andajos) [inline CONTRIBUTING.md:23]: @2ynn 
How about using imperative form in commit messages? Like `Fix some-bug` instead of `Fixed`. As that's how git makes commit messages when using git merge, same for github iirc.

Maybe if it's on this form: `Add/Refactor/Test/Docs/Remove commit-message`.
You and tristan write commits in different ways though so it depends on your convention.

Maybe also **not** use `[#issue-number] commit-message`, because we can reference issues in PRs.

You can check [this freecodecamp article](https://www.freecodecamp.org/news/writing-good-commit-messages-a-practical-guide/) too if would like to. There are some useful guidelines there too.

Tell me what you think of my suggestion and I'll add some edits.

**Comment by mina-andajos** [comment]
Also a suggestion, just a preference of mine. How about we add a pre-commit hook to format the python code? Just to make everything look coherent. I can add something like [this](https://github.com/psf/black/blob/main/.pre-commit-hooks.yaml) and reference a guide (like [this one](https://www.youtube.com/watch?v=psjz6rwzMdk) on how to set it up for contributors.

File: CONTRIBUTING.md
Comment (2ynn) [reply CONTRIBUTING.md:23]: oh yes my commit messages are terrible, i definitely need guidelines ;) yes imperative form is good! 
ok for not including issue number in the PR title and referencing it in the comment section. To me having the issue number in commits though could help when looking back at the git log history. But maybe it doesn't make sense.

**Comment by tristanlatr** [comment]
In my opinion, pull requests names, branch names and commit messages are largely aesthetic, and should not be constrained at this point of development.

What is important though, is having reviews and approvals all all PR that are open before merging them. Linking the issue that the PR is closing is also useful.

File: CONTRIBUTING.md
Comment (mina-andajos) [reply CONTRIBUTING.md:23]: Also how about adding [WIP] before a PR's name to let you know it's not finished yet and tell the contributors to mention you, share screenshots (I could edit the PR template to include maybe screenshots, links to pages, bullet points to what they did exactly, etc...)

**Comment by 2ynn** [comment]
I don't think is just aesthetics, at least on this repo (agree about not needing to put constraints on contributers' branch names on their own forks).  But for example when we need to clean up old branches it would be useful to quickly see which issue they addressed.

@mina-andajos I'd be down to switch to black but I've never set it up, i still use `yapf` as a general guideline and I don't automatically format on save.

**Comment by 2ynn** [comment]
> What is important though, is having reviews and approvals on all PR that are open before merging them. Linking the issue that the PR is closing is also useful.

@tristanlatr ok I added a protection on develop branch so all pull request need at least one approval

File: CONTRIBUTING.md
Comment (2ynn) [reply CONTRIBUTING.md:23]: can we simply use the `Draft` functionality for marking PRs as not ready?

**Comment by mina-andajos** [comment]
> @mina-andajos I'd be down to switch to black but I've never set it up, i still use `yapf` as a general guideline and I don't automatically format on save.

We could setup [yapf](https://github.com/google/yapf/blob/main/plugins/pre-commit.sh) instead if that's what you use here. It'll just format all the staged changes automatically before committing. Whether the contributor use it in their editor or not. I can add the pre-commit and write instructions for contributors to set it up.

**Comment by 2ynn** [comment]
> We could setup [yapf](https://github.com/google/yapf/blob/main/plugins/pre-commit.sh) instead if that's what you use here. It'll just format all the staged changes automatically before committing. Whether the contributor use it in their editor or not. I can add the pre-commit and write instructions for contributors to set it up.

We haven't agreed on any formatters so far, so I'd be okay testing black. And yes as long as it only formats the actual changes and not the whole files I think that's good (not that we don't need general reformatting but let's not mess with the whole codebase for now)

File: CONTRIBUTING.md
Comment (tristanlatr) [reply CONTRIBUTING.md]: Yes using draft sounds better.

**Review by tristanlatr — APPROVED** [review]
Great guidelines, thanks.

**Review by mina-andajos — APPROVED** [review]
Thanks for your efforts.

**Pull request merged**
